crates/ide_db/src/helpers.rs

```diff
@@ -190,14 +188,16 @@ pub fn for_each_tail_expr(expr: &ast::Expr, cb: &mut dyn FnMut(&ast::Expr)) {
                 }
             }
         }
-        ast::Expr::LoopExpr(l) => {
-            for_each_break_expr(l.label(), l.loop_body().and_then(|it| it.stmt_list()), &mut |b| {
-                cb(&ast::Expr::BreakExpr(b))
-            })
-        }
+        ast::Expr::LoopExpr(l) => for_each_break_and_continue_expr(
+            l.label(),
+            l.loop_body().and_then(|it| it.stmt_list()),
+            &mut |b| cb(&b),
+        ),
```

Comment:
This is incorrect, `continue` is not to be considered a tail expression as it doesn't break off a loop